monitoring/src/supply-monitor.ts

```diff
@@ -0,0 +1,112 @@
+import { SystemEventType } from "./system-event"
+
+import type { BigNumber } from "ethers"
+import type { TBTCToken } from "@keep-network/tbtc-v2.ts/dist/src/chain"
+import type { Monitor as SystemEventMonitor, SystemEvent } from "./system-event"
+
+// The block span the supply change is checked for. It is 12 hours expressed
+// in ETH blocks, assuming 12 seconds per block on average.
+const totalSupplyBlockSpan = (12 * 60 * 60) / 12
+
+const totalSupplyChangeThreshold = 10 // 10%
+
+const HighTotalSupplyChange = (
+  difference: BigNumber,
+  change: BigNumber,
+  threshold: number,
+  referenceBlock: number,
+  currentBlock: number
+): SystemEvent => ({
+  title: "High TBTC token total supply change",
+  type: SystemEventType.Critical,
+  data: {
+    change: `${difference.gte(0) ? "+" : "-"}${change.toString()}%`,
+    threshold: `${threshold}%`,
+    referenceBlock: referenceBlock.toString(),
+  },
+  block: currentBlock,
+})
+
+export interface SupplyMonitorPersistence {
+  lastHighTotalSupplyChangeBlock: () => Promise<number>
+  updateLastHighTotalSupplyChangeBlock: (block: number) => Promise<void>
```

Comment:
Could be worth documenting those two functions in two sentences. 

My understanding of `lastHighTotalSupplyChangeBlock` is that it represents the end block of the check when the alert of the high total supply change` was triggered the last time. If the alert was never triggered, this is 0.

Is this correct?

Reply:
Yep. Added a comment faea9554